txprocessor/txprocessor.go

```diff
@@ -190,6 +190,12 @@ func (tp *TxProcessor) ProcessTxs(coordIdxs []common.Idx, l1usertxs, l1coordinat
 		if err == nil {
 			err = tp.s.MakeCheckpoint()
 		}
+		go func() {
```

Comment:
I think it can be dangerous to have "untracked" goroutines as a general practice.  For example, in this case you could stop the node at the middle of a `DeleteOldCheckpoints`, and then one of the old checkpoints gets half deleted, leading to possible unexpected results.  In this case I think it's not severe, because old checkpoints are not supposed to be read any time (that's why they are deleted).  But I think it's a better practice to keep track of all the go routines and make sure they are stopped on termination of the program.

I don't have the perfect solution for this because the StateDB is synchronous by nature, so it doesn't have `start` or `stop` methods, but here's an idea:
- Leave the `DeleteOldCheckpoints` inside of `MakeCheckpoint` in KVDB but run it via a goroutine.
- In KVDB add a waitgroup.  Before starting the gorotuine call `wg.Add(1)`.  And once the `DelteOldCheckpoints` is done call `wg.Done()`.
- In `KVDB.Close()` call `wg.Done()`.

Reply:
Good catch. Please check the updated implementation.